streams/src/test/java/org/apache/kafka/test/KStreamTestDriver.java

```diff
@@ -184,9 +185,15 @@ public void process(final String topicName, final Object key, final Object value
 
     private ProcessorNode sourceNodeByTopicName(final String topicName) {
         ProcessorNode topicNode = topology.source(topicName);
-
-        if (topicNode == null && globalTopology != null) {
-            topicNode = globalTopology.source(topicName);
+        if (topicNode == null) {
+            for (final String sourceTopic : topology.sourceTopics()) {
+                if (Pattern.compile(sourceTopic).matcher(topicName).matches()) {
```

Comment:
Does it make sense to store compiled pattern so that the matching can be checked quicker ?

Reply:
Not sure. This is only testing code.

Reply:
To my understanding, some users would base their code on this driver code.

Reply:
Well, but wouldn't this be testing code they write... What usage would `KStreamTestDriver` have in production code?

Btw: `KStreamTestDriver` will be removed shortly anyway as we now have `TopologyTestDriver` as public test-driver class.